ebs/ebs.go

```diff
@@ -303,6 +303,12 @@ func (d *Driver) CreateVolume(req Request) error {
 			log.Debugf("Failed to update tags for volume %v, but continue", volumeID)
 		}
 		volume.PreExist = true
+
+		// detach from all instances
+		for _, attachment := range ebsVolume.Attachments {
```

Comment:
One way is to check the VM state before detaching, either terminated or stopped, otherwise fail the operation.